### DISCUSSION TITLE
osu!lazer would start with a more simplistic menu (menu skinning)(game intro idea)

### DISCUSSION THREAD

**Discussion by oliverlundstrom** [ppy/osu#29221] [Ideas]
I think the main menu looks overall very complicated and I have played on stable for many years now, I can only imagine how a new user feels.
A fix would be a way to disable the menus from the top, disable the notification icon and menu and put it into the settings, disable it asking you to log in instantly in the code somehow, making the menu elements customizeable even.

If it was up to me I would make a first startup that had an intro, it would first ask you how much you know about osu.
It would have options for:
- new (gives you full intro and tutorial that I explain later and full setup help)
- knows about the game (gives tutorial and helps with setup)
- has used for a long time (lets the user instantly into the main menu

if you press:
NEW PLAYER
- first helps you set up tablet or mouse and keybindings.
- could have basic display setting menu
- then has menu for gamemode, you can select many and it could give tutorials on the ones you selected interest in, also this selection could be changed and used for something, for example if I choose I only play osu standard and dont want any other maps to take space, the game would automatically delete all other game mode maps for example, but it could also just not show them until you enable that mode and you can always just have a setting for how it was before.
- assumes certain set of default settings for new players. Game should look simplistic, not animated and full of particles for new players especially (100% bg dim by default and no hit lightning). The fun part about the game is getting better and that could be echoed in the next part
- has a tutorial that explains the point of the game, its almost like digital dancing. It would explain the basics of gameplay, it would then explain how maps are made and where to get them and give some understanding on the history, culture and community of osu, since that is the lifeblood of this game.
- The game could test you on how good you are if you are new, then remembering this difficulty to recommend maps in next section.
- Then it could have a menu where you can choose some map sets to download since new players always have too little maps. It could ask you what kind of music you like and could also just download some featured artist music if they click that they want more maps (it should be recommended since you get better by playing different maps not the same map over and over again)
- Then the game would be normal BUT the menu would be different, ill explain how in the next section of this.

OLD PLAYER
the menu could just be a circle. When you click it the top bar would have only a music player on the top right so you can stop the song and the classic stable menu would pop up with just 4 choises. play edit settings and exit.
In the setting at the top could be login page.
then under that could be the menu setup.

menu skins:
- make the buttons in the menu movable when making the skin. 
- allow for changing of all elements in the game and adding of a lot of new ones.
- disabling animations.

**Comment by oliverlundstrom** [comment]
if not fully customizeable skins, make a section in the settings for enabling parts of the menu. That could be easier to use and implement right now, but menu skins, where you can just customize everything, would be sick.

**Comment by Stoppedpuma** [comment]
> A fix would be a way to disable the menus from the top,

Ctrl + Shift + T

> disable it asking you to log in instantly in the code somehow

This is a security feature and should only appear when you login, this will not change.

> making the menu elements customizeable even + menu skins

More in-depth client skinning is planned, you can probably find what you're looking for by searching around.

> If it was up to me I would make a first startup that had an intro, it would first ask you how much you know about osu.
It would have options for:

I think this is rather unnecessary as the setup screen is a ~30 second process for most players.

**Reply by oliverlundstrom** [reply]
as I said, the application looks way too complicated. If you dont teach you wont get more players. You really should read the post before commenting.

**Reply by oliverlundstrom** [reply]
If you create more before planning for the end product a lot of work will be for nothing no?

**Reply by Stoppedpuma** [reply]
I clearly did read the post if I'm directly responding to your own words.

> as I said, the application looks way too complicated. If you dont teach you wont get more players.

I'm confused on how buttons with labels such as "Next" or "Play" or "Mods" are confusing? There's also a tutorial beatmap which teaches gameplay so in my eyes there is no confusion. If you can explain how buttons labelled with what they do is complicated that would be appreciated.

![image](https://github.com/user-attachments/assets/410ec4f7-0a8d-4981-ba4d-a5c83d34b661)


> If you create more before planning for the end product a lot of work will be for nothing no?

I'm not even sure what this means fully, osu! already has more upcoming designs which are being implemented. There's no "lot of work for nothing" when the majority of the players like the interface. Unless you're coming from a platform such as Reddit where everyone just complains anyways there's an extreme lack of complaints, even then though the majority on Reddit are in favour of the design.

**Reply by oliverlundstrom** [reply]
also 
-how tf is asking you to login a security feature?
-I heard about in depth skinning and didnt like what I heard. Havent seen anything about it on roadmap when I looked or in ideas but maybe I am blind.
-how is ctrl shift t an acceptable answer in your opinion to just completely curb what I am saying with it like its a trump card. Why isnt this a setting you can switch why isnt it clearer why was it added if its hidden and badly implemented I dont understand.
-what about new players do you not understand?

**Reply by oliverlundstrom** [reply]
the top and sides and bottom of the screen is filled with things happening for no reason

**Reply by oliverlundstrom** [reply]
sorry if I just am* rude but man I hate feeling misunderstood yet I fucking blow at writing.

**Reply by Stoppedpuma** [reply]
> how tf is asking you to login a security feature?

How is it not? You're complaining about a 2FA code which is a security step implemented in basically every major platform.

> I heard about in depth skinning and didnt like what I heard. Havent seen anything about it on roadmap when I looked or in ideas but maybe I am blind.

There are higher priorities than making your skin work in x menu.

> Why isnt this a setting you can switch why isnt it clearer why was it added if its hidden and badly implemented I dont understand.

You clearly didn't look hard enough as it's in plain sight inside of controls.

![image](https://github.com/user-attachments/assets/fe8f22a3-6848-4859-be91-8e09ac9d682c)

> what about new players do you not understand?

Giving me blind complaints of how I don't understand new players is not a correct way to voice your opinion as I can't comment on anything if you aren't giving me something to comment on. If I'm going to take a broad guess I'm going to say you're complaining about me saying the interface is simple. If you can explain as I said above how buttons labelled with what they do is confusing.

> the top and sides and bottom of the screen is filled with things happening for no reason

Can you explain where you're seeing this?

**Reply by oliverlundstrom** [reply]
- I am complaining about the menu popping up for no reason while starting the application.
- are there higher priorities than having a good plan for the future?
- My exact point, Why Is It There and not as a setting?
- my complaints are not blind, if you read what I said or if I didnt write it so poorly, you would have understood that my issue was that this application and everything to do with the whole game and community is insanely complicated, saying as a person playing this game for 7 + years and who has taught 10 + people to play this game it is not easy to understand, I dont see much attempt to simplify it.

**Reply by oliverlundstrom** [reply]
and also specifically to do with the art direction and the lack of ability to roll back all the new stuff into how it was, or just less flashy

**Reply by oliverlundstrom** [reply]
![osu_2024-08-01_03-52-06](https://github.com/user-attachments/assets/ee3e88c0-d8f9-48ef-b17e-4b56de329326)
there are 23 buttons on the fucking main menu
How is that not overly complex to you?

**Reply by oliverlundstrom** [reply]
sorry for the tone idk why it sounds so sharp :( was tryna be funny a little

**Reply by Stoppedpuma** [reply]
> I am complaining about the menu popping up for no reason while starting the application.

Going to assume you mean the setup menu, simply complete it and it goes away.

> are there higher priorities than having a good plan for the future?

I'm not sure what this means? If you mean skinning there's a lot more of concern then PNGs working on every menu.

> My exact point, Why Is It There and not as a setting?

Because it's unnecessary, it shows information which is useful to the user and is apart of making the client be all together instead of having to use the website for half the features of the game. If you want a toggle you're going to have to win a war with bdach as he also agrees a toggle for everything is unnecessary and just more work on the maintaining side of things.

> everything to do with the whole game and community is insanely complicated, saying as a person playing this game for 7 + years and who has taught 10 + people to play this game it is not easy to understand, I dont see much attempt to simplify it.

If it means anything at all a "story mode" is in the works which will introduce players to the game,

> and also specifically to do with the art direction and the lack of ability to roll back all the new stuff into how it was, or just less flashy

There is no "Roll back all the new stuff into how it was", lazer is a complete rewrite and has been said about a thousand times that the UI will not match stable. Changing the entire design to comply with the 1% is off the table as well. Since you seem to complain that lazer is different than stable in your other post as well, you may prefer to just stay on stable as lazer is all about change.

**Reply by oliverlundstrom** [reply]
Why are you assuming, when the context is there to fully understand? 

I said I dont understand why the menu asking you to login needs to pop up, and now you think I am talking about setup window?

you are not sure what a good plan means? 

the ui could easily match stable just remade. Your arguments make no sense and you have not answered a single thing.

**Reply by oliverlundstrom** [reply]
the menu asking you to login, on top of everything else bullshit happening on the screen. Its a horrible first impression.

**Reply by oliverlundstrom** [reply]
you seem like one of these people who dont understand what critizism is and if so, you dont need to continue replying to me this is going nowhere.

**Reply by Stoppedpuma** [reply]
> Why are you assuming, when the context is there to fully understand?

Because you're blindly just pointing "this!", "that!" and not saying what this and that are. While I may just be skipping over words since it's very late where I am in which case I'm sorry, I would still appreciate more in-depth explanations about complains rather than "too many buttons" or something around those lines.

> I said I dont understand why the menu asking you to login needs to pop up, and now you think I am talking about setup window?

Dude just login then, it's required for 80% of the games functionality, I don't know what more you want me to say here?

> you are not sure what a good plan means?

It's 03:00 and I'm trying to read half broken english, cut me a break and instead mention directly what you are complaining about so I'm not guessing constantly. E.g "Is there a good plan ahead for skinning menus".

> the ui could easily match stable just remade. Your arguments make no sense and you have not answered a single thing.

Allow me to rephrase, paraphrasing the words of main dev team "stable and lazer are two different clients, lazer will not look like stable."

> you seem like one of these people who dont understand what critizism is and if so, you dont need to continue replying to me this is going nowhere.

Kindly, fuck off. You came to a discussion board and are engaging in discussions you opened, there's no room for criticism on your part here. Additionally how are you going to say I don't understand criticism when I'm simply asking for you to give more in-depth explanations of how something is wrong in your eyes? That is literally listening to what you're complaining about. I've done nothing but ask for better examples and state words directly from the main development team.

**Reply by oliverlundstrom** [reply]
You kindly fuck off man, nothing u said has reflected anything I said and you make no sense. You just come here, assume im an idiot and when I am not you become defensive for no reason. I wasnt talking to you at all. 
"just log in" is a great example of your hubris. I am talking about the new user experience for fucks sake

**Reply by oliverlundstrom** [reply]
you for some reason think user experience and ideas is not worth sharing here and if so I apologize but you have not said one word that had regarded me as a thinking human yet, if not for the snarky remarks I would think you were ai.

**Reply by oliverlundstrom** [reply]
so forgive me for feeling attacked when someone comes here and gives me "advice" that is just the most basic dogshit I have ever seen and disregards everything interesting about what I typed
edit: well that was the other post but here you just ignore what I am trying to say. You just assume im dumb for some reason I feel.

**Reply by oliverlundstrom** [reply]
also its 4:25 for me bud

**Reply by oliverlundstrom** [reply]
also the fact you say there is no room for criticism on a discussion forum just confirms you or I dont understand what it is. I dont get why you people insist on taking everything personally, if I say: "hey this game sucks", you should read that as "this person thinks this game sucks", not as, "this game sucks". It is merely an opinion and if you cant see the vision I put forward in the intro idea, why are you commenting?

**Reply by oliverlundstrom** [reply]
you could say that its unclear and I would not take that personally but would probably apologize and try to make it better, but you assume.

**Reply by Stoppedpuma** [reply]
> You just come here, assume im an idiot

Your words, not mine. I haven't called you an idiot anywhere, I'm simply here to discuss what you find confusing.

> nothing u said has reflected anything I said and you make no sense.

How do you figure? I'm directly responding to what you point out.

> I wasnt talking to you at all.

What?
![image](https://github.com/user-attachments/assets/2000e2b1-0e37-4042-8cff-eee58e6f7e5b)

> "just log in" is a great example of your hubris. I am talking about the new user experience for fucks sake

This is literally expected behaviour? Try logging out and look at any online facing menu, it requires login. Take any other major online game as an example, what's the first thing they ask before you start playing? Most likely to login.

> you for some reason think user experience and ideas is not worth sharing here

A large portion of my discussion on this git repo is for pushing new ideas forward. Why you seem to be offended is beyond me at this point.

> you have not said one word that had regarded me as a thinking human yet

This applies to the above, I see no reason for you to be offended? I've simply asked you to explain further, in this context further means how you would change it or why it feels bad, etc.

**Reply by Stoppedpuma** [reply]
> you could say that its unclear and I would not take that personally but would probably apologize and try to make it better, but you assume.

Assuming is a way of saying something is unclear, no?

**Reply by oliverlundstrom** [reply]
do you just copy the thing ur quoting or how do you do that?

**Reply by oliverlundstrom** [reply]
Im sorry, I think I shouldnt have made this post when this sleep deprived.

**Reply by oliverlundstrom** [reply]
ig the comment you left felt constricting the way you answered my thing, but I dont know if I am right at all so idk, I just feel annoyed at myself for wasting ur time for no reason.

**Reply by oliverlundstrom** [reply]
I do still very much think I have a point but I should probably make like a youtube video about it or something

**Reply by Stoppedpuma** [reply]
> do you just copy the thing ur quoting or how do you do that?

Simply putting > before it like so:

![image](https://github.com/user-attachments/assets/6fa079fc-4584-40ab-9e0e-ca8624406b2d)

> Im sorry, I think I shouldnt have made this post when this sleep deprived.

No worries, we both seem against each other possibly for plain ignorance / misunderstandings from both of us being awake this long.

> ig the comment you left felt constricting the way you answered my thing, but I dont know if I am right at all so idk, I just feel annoyed at myself for wasting ur time for no reason.

Don't worry about it, a lot of your criticism is possibly correct and I'm just missing it at the time of writing, english is my second language and I've been up far too long so forgive me for anything which may have seemed as an attack towards you.

**Reply by oliverlundstrom** [reply]
im glad we could come to peace, I hope you can get some sleep ill try right god damn now :I
english is my second language aswell so sorry for it being kinda broken, also have dyslexia so sometimes it just becomes fucked and I dont even notice.
>also thank you! <3 :D

**Reply by oliverlundstrom** [reply]
one more thing tho, should I just close this do u recon?

**Reply by oliverlundstrom** [reply]
how do you delete these?

**Reply by Stoppedpuma** [reply]
Depends on what you feel, If you think my point of saying that lazers UI won't match stable and will be more advanced is an answer then sure. Otherwise you can keep it open and see if anyone else has some words, I wouldn't hold my breath on the design becoming much more simple than it is now though.

**Reply by Stoppedpuma** [reply]
> how do you delete these?

I don't think you can but I believe this button should hide it? Otherwise deleting comments, editing the main post and deleting revision history is a workaround.
![image](https://github.com/user-attachments/assets/cb688aff-301e-42b2-be54-576ecb817800)

**Reply by oliverlundstrom** [reply]
Hmm well ill try to convey how I feel on a video like I actually really feel I have enough to say for one, I think its good to have differing opinions thrown about, someone playing the devils advocate, even if its not as prevalent. Because in my opinion thats the way to get the best results, I do actually just want to help. I'll try to think on this convo about how to convey the points, thanks for being patient with me 🙏
